Code/Framework/AzFramework/AzFramework/Quality/QualityCVarGroup.cpp

```diff
@@ -0,0 +1,244 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzFramework/Quality/QualityCVarGroup.h>
+#include <AzFramework/Quality/QualitySystemBus.h>
+
+#include <AzCore/Settings/SettingsRegistryImpl.h>
+#include <AzCore/Settings/SettingsRegistryVisitorUtils.h>
+#include <AzCore/std/smart_ptr/unique_ptr.h>
+#include <AzCore/Console/IConsole.h>
+#include <AzCore/StringFunc/StringFunc.h>
+
+namespace AzFramework
+{
+    QualityCVarGroup::QualityCVarGroup(AZ::SettingsRegistryInterface* registry, AZ::IConsole* console, AZStd::string_view name, AZStd::string_view path)
+        : m_console(console)
+        , m_numQualityLevels(0)
+        , m_path(path)
+        , m_qualityLevel(0)
+        , m_settingsRegistry(registry)
+    {
+        AZ_Assert(registry, "QualityCVarGroup requireds a valid settings registry.");
+        AZ_Assert(console, "QualityCVarGroup requireds a valid settings registry.");

```

Comment:
typo: Two issues
1. Both asserts have a typo in the word "requireds".
2. The second assert should mention that a valid AZ::Console is required.

_However_, I think a better solution is to have the constructor accept both the SettingsRegistryInteface and IConsole by reference, so that user is force to pass in a valid instance and perform the nullptr check on their end.

Reply:
I was doing that originally for that very reason, but don't remember why I changed it to a pointer.  Will post back here if see what the issue was - hopefully it wasn't some template compilation issue